src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,195 @@
+
+pub struct Clickhouse {
+    config: ClickhouseConfig,
+}
+
+#[async_trait::async_trait]
+impl Connector for Clickhouse {
+    async fn create_sink(
+        &mut self,
+        sink_context: SinkContext,
+        builder: SinkManagerBuilder,
+    ) -> Result<Option<SinkAddr>> {
+        let db_url = self.connection_url();
+        let sink = ClickhouseSink { db_url, pool: None };
+        builder.spawn(sink, sink_context).map(Some)
+    }
+
+    fn codec_requirements(&self) -> CodecReq {
+        CodecReq::Structured
+    }
+}
+
+impl Clickhouse {
+    fn connection_url(&self) -> String {
+        let url = self.config.url.as_str();
+        let compression = self.config.compression.unwrap_or_default();
+
+        format!("{url}?compression={compression}")
+    }
+}
+
+#[derive(Deserialize)]
+struct ClickhouseConfig {
+    url: Url<ClickHouseDefaults>,
+    compression: Option<Compression>,
+    database: String,
+    columns: Vec<Column>,
+}
+
+impl ClickhouseConfig {
+    const DEFAULT_PORT: u16 = 9000;
+}
+
+impl ConfigImpl for ClickhouseConfig {}
+
+#[derive(Clone, Copy, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum Compression {
+    None,
+    Lz4,
+}
+
+impl Default for Compression {
+    fn default() -> Compression {
+        Compression::None
+    }
+}
+
+impl Display for Compression {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Compression::None => "none",
+            Compression::Lz4 => "lz4",
+        }
+        .fmt(f)
+    }
+}
+
+#[derive(Deserialize)]
+struct DatabaseColumns {
+    #[serde(flatten)]
+    columns: Vec<Column>,
+}
+
+#[derive(Deserialize)]
+struct Column {
+    name: String,
+    #[serde(rename = "type")]
+    type_: ClickHouseType,
+}
+
+#[derive(Deserialize)]
+enum ClickHouseType {
+    UInt8,
+}
+
+pub(crate) struct ClickhouseSink {
+    db_url: String,
+    pool: Option<Pool>,
+}
+
+#[async_trait::async_trait]
+impl Sink for ClickhouseSink {
+    async fn connect(&mut self, _ctx: &SinkContext, _attempt: &Attempt) -> Result<bool> {
+        let pool = Pool::new(self.db_url.as_str());
+        self.pool = Some(pool);
+
+        Ok(true)
+    }
+
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        event: Event,
+        _ctx: &SinkContext,
+        _serializer: &mut EventSerializer,
+        _start: u64,
+    ) -> Result<SinkReply> {
+        let mut client = self
+            .pool
+            .as_ref()
+            .ok_or_else(|| Error::from(ErrorKind::NoClickHouseClientAvailable))?
+            .get_handle()
+            .await?;
+
+        // TODO: fetch column data from the context?
+        let columns = get_column_data();
+
+        let block = columns
+            .into_iter()
+            .fold(Block::new(), |block, (name, ty)| match ty {
+                ClickHouseType::UInt8 => add_uint8_column(block, name, &event),
+            });
+
+        client.insert("people", block.clone()).await?;
+
+        Ok(SinkReply::ACK)
```

Comment:
As the sink is `auto_ack: true` you don't need to explicitly ACK. The runtime will handle this for you.

```suggestion
        Ok(SinkReply::NONE)
```